aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputFile.java

```diff
@@ -29,7 +30,11 @@
 
 public class S3OutputFile extends BaseS3File implements OutputFile {
   public S3OutputFile(S3Client client, S3URI uri) {
-    super(client, uri);
+    this(client, uri, AwsCatalogProperties.S3_SSE_TYPE_NONE, null, null);
+  }
+
+  public S3OutputFile(S3Client client, S3URI uri, String sseType, String sseKey, String sseMd5) {
```

Comment:
As I mentioned in the comment above, this is what I think we want to try to avoid where we have to add constructor params per property we need to add.  There are already a few others we need to push through (multipart size, canned acl, etc.).  I think turning the `AwsProperties` into a wrapper would help as we expand this functionality.

Reply:
Yes I agree it sounds better to have all configuration in the separated class. Let me update it.